src/utils/polling.js

```diff
@@ -121,46 +100,67 @@ function addPollToDb(resource) {
  * @param {Subscription} subscription - Subscription resource
  */
 async function initialPoll(subscription) {
-  const topicName = getTopicName(subscription);
-  if (!topicName) {
-    logger.info('Initial poll not needed for subscription without topic.');
-    return;
-  }
-
-  const resourceToPoll = namedEventToResourceType(topicName);
-
-  // Don't poll if resource is already being polled for.
-  const mostRecentPoll = db.select('polling', (p) => p.resource === resourceToPoll);
-  if (mostRecentPoll.length > 0) {
-    logger.info(`${resourceToPoll} resource is already being polled.`);
-    return;
-  }
+  const subscriptionTopic = getSubscriptionTopic(subscription);
 
+  // Make the fhirClient
   const accessToken = await getAccessToken(fhirClientConfig);
   const options = {
     baseUrl: fhirClientConfig.baseUrl,
     auth: { bearer: accessToken },
   };
-
   const fhirClient = mkFhir(options);
-  fhirClient
-    .search(getSearchQuery(resourceToPoll))
-    .then((response) => {
-      const { data } = response;
-      addPollToDb(resourceToPoll);
-
-      // Store fetched resources in local database
-      if (data.total > 0) {
-        logger.info(`Storing ${data.total} fetched resource(s) for ${resourceToPoll} into database.`);
-        const resources = data.entry.map(entry => entry.resource);
-
-        resources.forEach((resource) => {
-          const collection = `${resource.resourceType.toLowerCase()}s`;
-          db.upsert(collection, { id: resource.id }, r => r.id === resource.id);
-        });
-      }
-    })
-    .catch((err) => logger.error(err));
+
+  subscriptionTopic.resourceTrigger.forEach((resourceTrigger) => {
```

Comment:
This was failing because the check was coming back `null` so it was easy enough to perform the check on the create new subscription and then reject if the topic was invalid